packages/axes/src/types.ts

```diff
@@ -0,0 +1,122 @@
+import { SpringValues } from 'react-spring'
+import {
+    ScaleBand,
+    ScaleLinear,
+    ScaleOrdinal,
+    ScalePoint,
+    ScaleTime,
+    ScaleSymLog,
+    ScaleLogarithmic,
+} from 'd3-scale'
+import React from 'react'
+
+export type GridValuesBuilder<T> = T extends number
+    ? number[]
+    : T extends string
+    ? string[]
+    : T extends Date
+    ? Date[]
+    : never
+
+export type GridValues<T extends string | number | Date> = number | GridValuesBuilder<T>
+
+export type Point = {
+    x: number
+    y: number
+}
+
+export type ScaleWithBandwidth =
+    | (ScaleBand<any> & { type: 'band' })
+    | (ScalePoint<any> & { type: 'point' })
+
+export type AnyScale =
```

Comment:
Maybe this type can be a generic where we pass the type of the value.

Reply:
I coudln't get this to work.